src/_pytest/cacheprovider.py

```diff
@@ -201,19 +199,27 @@ def set(self, key: str, value: object) -> None:
             )
         else:
             with f:
-                f.write(data)
-
-    def _ensure_supporting_files(self) -> None:
-        """Create supporting files in the cache dir that are not really part of the cache."""
-        readme_path = self._cachedir / "README.md"
-        readme_path.write_text(README_CONTENT, encoding="UTF-8")
+                json.dump(value, f, ensure_ascii=False, indent=2)
 
-        gitignore_path = self._cachedir.joinpath(".gitignore")
-        msg = "# Created by pytest automatically.\n*\n"
-        gitignore_path.write_text(msg, encoding="UTF-8")
+    def _ensure_cache_dir_and_supporting_files(self) -> None:
+        """Create the cache dir and its supporting files."""
+        self._cachedir.mkdir(exist_ok=True, parents=True)
 
-        cachedir_tag_path = self._cachedir.joinpath("CACHEDIR.TAG")
-        cachedir_tag_path.write_bytes(CACHEDIR_TAG_CONTENT)
+        files: Iterable[tuple[str, str | bytes]] = (
```

Comment:
Not sure how CI passes, but anyway I think we have to keep using `Tuple` and `Union` for a bit

```suggestion
        files: Iterable[Tuple[str, Union[str, bytes]]] = (
```

Reply:
Done.